lib/balanced_vrp_clustering.rb

```diff
@@ -134,15 +187,96 @@ def build(data_set, cut_symbol, cut_ratio = 1.0, options = {})
         self
       end
 
+      def move_limit_violating_dataitems
+        @needed_moving = @items_with_limit_violation.size
```

Comment:
Since we are "pop"ing out stuff from the `@items_with_limit_violation` array, its size become zero inside this function so we need to register its size for using it later in the stopping condition.

Reply:
Renamed `@needed_moving` into `@limit_violation_count`, it is more appropriate